File: C# Pattern Programs/Wave Patterns/Pattern 24.cs
using System;

   class Wave_24
 {
    public static void Main(string[] args)
    {

    int wave_height = 5; //change  value to increase or decrease the height of wave
    int wave_length = 4; //change  value to increase or decrease the length of wave

    int i;
    int j;
    int k;



    for (i = 1;i <= wave_height;i++)
    {
    for (j = 1;j <= wave_length;j++)
    {
    if (i != wave_height)
    {
    Console.Write("##");
    }
    else
    {
    Console.Write("  "); // 2 spaces
    }

    for (k = 1;k <= 4;k++)
    {
    if (i == 1 || k == 1 || k == 4)
    {
    Console.Write("  ");
    }
    else
    {
    Console.Write("##");
    }
    }

    if (i != wave_height)
    {
    Console.Write("##");
    }
    else
    {
    Console.Write("  "); // 2 spaces
                }
    }

    Console.WriteLine();
    }

     Console.ReadKey(true);
    }
 }

File: PythonPatternPrograms/WavePatterns/Pattern 24.py
wH = 4  # 1
wL = 4  # 2

for x in range(1, wH + 1):
    for y in range(1, wL + 1):
        if (x != wH):
            print("##", end="")
        else:
            print("  ", end="")  # 3
        for z in range(1, 5):
            if (x == 1 or z == 1 or z == 4):
                print("  ", end="")  # 3
            else:
                print("##", end="")

        if (x != wH):
            print("##", end="")
        else:
            print("  ", end="")  # 3
    print()

    """
1) wH - change value to increase/decrease the height of the wave
2) wL - change value to increase/decrease the length of the wave
3) 2 whitespaces

    """